### ISSUE TITLE
Make subsequent unpacking of `Any` faster

### ISSUE THREAD

**Issue by alexander-yevsyukov** [SpineEventEngine/core-java#1267]
There are cases when we repeatedly invoke `AnyPacker.unpack()` on an `Any` instance. This happens, for example, when we obtain a type of a `Signal` that we're processing.

Even though, `Any` handles the caching for us, the `AnyPacker` still performs these operations:
```java
TypeUrl typeUrl = TypeUrl.ofEnclosed(any);
Class<? extends Message> messageClass = typeUrl.getMessageClass();
```
We can improve it by one of the following scenarios:
   Option 1. Look into the `Any`'s field `cachedUnpackValue` (which is `private`, BTW) and it's non-null, return its value. Since we don't pass the class when unpacking, having a value would mean the previous call for unpacking was successful. Since it's going to use Reflection, it may be a bit slow too.
  Option 2: Have a limited size cache which would map an instance of `Any` to extracted value.

**Comment by armiol** [comment]
See `AnyPacker.unpackFunc`. It does exactly that.

**Comment by alexander-yevsyukov** [comment]
After some more conversations with @armiol and looking into generated code, here's the another suggestion for implementation.

Extend Model Compiler with the ability to add a field that would cache the unpacked content of another field of type `Any` (and corresponding accessor method, of course). The name of the property and its type would come as options for the `Any` field. For example, for `Event` it would be:

```proto
message Event {
  //...
  google.protobuf.Any message = 2 [(required) = true, (validate) = true,
    (unpacked).name = "enclosed_message", 
    (unpacked).java_type = "EventMessage"];
  // ...
}
```

Using the `@@protoc_insertion_point(class_scope:spine.core.Event)` the Model Compiler would create:
  1. The field called `enclosedMessage`. The Java name for the field is calculated using standard conversion rules.
  2. The method `enclosedMessage()` which would populate the field, if it's `null` and return it, if not.

**Comment by alexander-yevsyukov** [comment]
Of course, the option `unpacked` makes sense only for `Any` proto fields. The compiler should generate a compile-time error when the option is set for a field of another type.
